subsys/net/lib/wifi_credentials/wifi_credentials.c

```diff
@@ -20,6 +22,82 @@ K_MUTEX_DEFINE(wifi_credentials_mutex);
 static char ssid_cache[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES][WIFI_SSID_MAX_LEN];
 static size_t ssid_cache_lengths[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES];
 
+int __wifi_creds_to_params(size_t argc, char *argv[],
```

Comment:
Why do we need variable args here? Can't we pass a simple `struct`? And also, except for Password/Passphrase others are straightforward, so, helper can be limited to that.